presto-main/src/test/java/com/facebook/presto/type/TestTimestampWithTimeZone.java

```diff
@@ -113,4 +133,41 @@ public void testCastToTimestamp()
                 TIMESTAMP,
                 sqlTimestampOf(2001, 1, 22, 3, 4, 5, 321, session));
     }
+
+    @Test
+    public void testCastInvalidTimestamp()
+    {
+        assertInvalidFunction("CAST('ABC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: ABC");
+        assertInvalidFunction("CAST('2022-01-00 00:00:00 UTC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: 2022-01-00 00:00:00 UTC");
+        assertInvalidFunction("CAST('AABC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: AABC");
+        assertInvalidFunction("CAST('2022-01-00 00:00:00 UTC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: 2022-01-00 00:00:00 UTC");
+        assertInvalidFunction("CAST('2022-01-01 00:00:00 ABC' AS TIMESTAMP WITH TIME ZONE)", INVALID_CAST_ARGUMENT, "Value cannot be cast to timestamp with time zone: 2022-01-01 00:00:00 ABC");
+    }
```

Comment:
This is all just testing that a plain string can't be converted to ts with tz right?

Reply:
The test testCastInvalidTimestamp() is checking to ensure that a string which does not follow the valid format for a timestamp with time zone cannot be cast to TIMESTAMP WITH TIME ZONE.

For each assertInvalidFunction call, an attempt is made to cast an incorrectly formatted string to TIMESTAMP WITH TIME ZONE, expecting each attempt to fail with an INVALID_CAST_ARGUMENT error.

The string 'ABC' is not a valid timestamp.
The string 'AABC' is not a valid timestamp.